### DISCUSSION TITLE
Issues with git worktrees?

### DISCUSSION THREAD

**Discussion by oysandvik94** [mfussenegger/nvim-jdtls#728] [General]
I was having some issues where the LSP would start, and diagnostics would work, but any action such as hover, goto def, etc would time out (in like 10 seconds). 

This was occuring in a repo that I had set up as a git worktree, while when I cloned the same repo normally, I had no issues. I tried recloning etc multiple times.

I'm not sure if it's because of worktrees, and I didn't have any time to debug it properly, but I was curious if anyone else had experienced issues with worktrees, or if this was just specific to my project?

**Comment by mfussenegger** [comment]
I never noticed any issues with worktrees, but also don't use them very often.

What could've happened is that there were some stale/corrupt project files in the directory that prevented jdtls from working (`.settings/` or `.project`) and a new clone got rid of them. A `git clean -xdff` would likely also have worked.

**Reply by oysandvik94** [reply]
That could be. There are actually some .project and .classpath files in the project already checked in, so maybe that messed it up. I'll try and git clean to see if it helps.

**Reply by oysandvik94** [reply]
It kinda just worked immediately hehe. Thank you!
